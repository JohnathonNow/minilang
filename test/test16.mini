fun f1(X) do
	print('f1({X})\n')
	if X = 1 then
		1
	elseif X % 2 = 0 then
		f1(X / 2)
	else
		f2(X)
	end
end

fun f2(X) do
	print('f2({X})\n')
	if X = 1 then
		1
	elseif X % 2 = 0 then
		f1(X)
	else
		f2(3 * X + 1)
	end
end

print('f1(123) = {f1(1234)}\n')
