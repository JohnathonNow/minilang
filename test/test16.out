f1(1234)
f1(617)
f2(617)
f2(1852)
f1(1852)
f1(926)
f1(463)
f2(463)
f2(1390)
f1(1390)
f1(695)
f2(695)
f2(2086)
f1(2086)
f1(1043)
f2(1043)
f2(3130)
f1(3130)
f1(1565)
f2(1565)
f2(4696)
f1(4696)
f1(2348)
f1(1174)
f1(587)
f2(587)
f2(1762)
f1(1762)
f1(881)
f2(881)
f2(2644)
f1(2644)
f1(1322)
f1(661)
f2(661)
f2(1984)
f1(1984)
f1(992)
f1(496)
f1(248)
f1(124)
f1(62)
f1(31)
f2(31)
f2(94)
f1(94)
f1(47)
f2(47)
f2(142)
f1(142)
f1(71)
f2(71)
f2(214)
f1(214)
f1(107)
f2(107)
f2(322)
f1(322)
f1(161)
f2(161)
f2(484)
f1(484)
f1(242)
f1(121)
f2(121)
f2(364)
f1(364)
f1(182)
f1(91)
f2(91)
f2(274)
f1(274)
f1(137)
f2(137)
f2(412)
f1(412)
f1(206)
f1(103)
f2(103)
f2(310)
f1(310)
f1(155)
f2(155)
f2(466)
f1(466)
f1(233)
f2(233)
f2(700)
f1(700)
f1(350)
f1(175)
f2(175)
f2(526)
f1(526)
f1(263)
f2(263)
f2(790)
f1(790)
f1(395)
f2(395)
f2(1186)
f1(1186)
f1(593)
f2(593)
f2(1780)
f1(1780)
f1(890)
f1(445)
f2(445)
f2(1336)
f1(1336)
f1(668)
f1(334)
f1(167)
f2(167)
f2(502)
f1(502)
f1(251)
f2(251)
f2(754)
f1(754)
f1(377)
f2(377)
f2(1132)
f1(1132)
f1(566)
f1(283)
f2(283)
f2(850)
f1(850)
f1(425)
f2(425)
f2(1276)
f1(1276)
f1(638)
f1(319)
f2(319)
f2(958)
f1(958)
f1(479)
f2(479)
f2(1438)
f1(1438)
f1(719)
f2(719)
f2(2158)
f1(2158)
f1(1079)
f2(1079)
f2(3238)
f1(3238)
f1(1619)
f2(1619)
f2(4858)
f1(4858)
f1(2429)
f2(2429)
f2(7288)
f1(7288)
f1(3644)
f1(1822)
f1(911)
f2(911)
f2(2734)
f1(2734)
f1(1367)
f2(1367)
f2(4102)
f1(4102)
f1(2051)
f2(2051)
f2(6154)
f1(6154)
f1(3077)
f2(3077)
f2(9232)
f1(9232)
f1(4616)
f1(2308)
f1(1154)
f1(577)
f2(577)
f2(1732)
f1(1732)
f1(866)
f1(433)
f2(433)
f2(1300)
f1(1300)
f1(650)
f1(325)
f2(325)
f2(976)
f1(976)
f1(488)
f1(244)
f1(122)
f1(61)
f2(61)
f2(184)
f1(184)
f1(92)
f1(46)
f1(23)
f2(23)
f2(70)
f1(70)
f1(35)
f2(35)
f2(106)
f1(106)
f1(53)
f2(53)
f2(160)
f1(160)
f1(80)
f1(40)
f1(20)
f1(10)
f1(5)
f2(5)
f2(16)
f1(16)
f1(8)
f1(4)
f1(2)
f1(1)
f1(123) = 1
